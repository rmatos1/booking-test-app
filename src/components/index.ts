export * from "./bookingDescription";
export * from "./bookingForm";
export * from "./images";
export * from "./inputGroup";
export * from "./navBar";
export * from "./styledComponents";
export * from "./testComponent";
