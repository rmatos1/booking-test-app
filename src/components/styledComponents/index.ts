export * from './buttons';
export * from './formComponents';
export * from './images';
export * from './layouts';
export * from './lists';
export * from './modalComponents';
export * from './texts';
