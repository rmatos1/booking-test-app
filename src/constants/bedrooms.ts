import { images } from '../assets';

export interface IBedroom {
  number: number;
  type: string;
  beds: {
    double?: number;
    single?: number;
  };
  img: string;
  fullCapacity: number;
  dailyPrice: number;
}

/**
 * constant that stores the maximum number of people possible in the biggest bedroom
 */
export const MAX_CAPACITY_BEDROOM = 4;

/**
 * constant with the available bedrooms to be booked
 */
export const hotelBedrooms: IBedroom[] = [
  {
    number: 1,
    type: 'Quadruple',
    beds: {
      double: 1,
      single: 2,
    },
    img: images.bedroom1,
    fullCapacity: 4,
    dailyPrice: 65,
  },
  {
    number: 2,
    type: 'Single',
    beds: {
      double: 1,
    },
    img: images.bedroom2,
    fullCapacity: 2,
    dailyPrice: 40,
  },
  {
    number: 3,
    type: 'Triple',
    beds: {
      double: 1,
      single: 1,
    },
    img: images.bedroom3,
    fullCapacity: 3,
    dailyPrice: 55,
  },
  {
    number: 4,
    type: 'Triple',
    beds: {
      single: 3,
    },
    img: images.bedroom4,
    fullCapacity: 3,
    dailyPrice: 50,
  },
  {
    number: 5,
    type: 'Single',
    beds: {
      single: 2,
    },
    img: images.bedroom5,
    fullCapacity: 2,
    dailyPrice: 35,
  },
];
