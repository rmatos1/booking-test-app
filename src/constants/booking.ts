export const initialBookingData = {
  checkIn: '',
  checkOut: '',
  qtyGuests: '',
  name: '',
  email: '',
  selectedBedroom: 0,
};

export const testBookingData = {
  id: '1',
  name: 'Julia',
  email: 'julia@teste.com',
  checkIn: '2023-12-19',
  checkOut: '2023-12-26',
  selectedBedroom: 3,
  qtyGuests: 3,
  totalPrice: 450,
};
