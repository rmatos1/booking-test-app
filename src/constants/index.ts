export * from './bedrooms';
export * from './booking';
export * from './styles';
