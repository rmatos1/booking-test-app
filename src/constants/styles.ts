/*
 * break points
 */
export const SMALL_DESKTOP = '990px';
export const TABLET = '768px';
export const MOBILE = '460px';

/***************************** */

export const HORIZONTAL_PADDING_DESKTOP = '25px';

export const HORIZONTAL_PADDING_MOBILE = '15px';
