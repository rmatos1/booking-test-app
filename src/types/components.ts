import { ReactNode } from "react";

export interface IComponentWithChildren {
  children?: ReactNode;
}

export interface IBaseDrawer extends IComponentWithChildren {
  isDrawerVisible: boolean;
  onCloseDrawer: () => void;
}
