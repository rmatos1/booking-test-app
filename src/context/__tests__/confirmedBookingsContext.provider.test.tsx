import { describe, test } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';

import { ReactNode } from 'react';
import { ConfirmedBookingsContext, IConfirmedBookingsContext } from '..';
import { TestComponent } from '../../components';
import { testBookingData } from '../../constants';

const testElement: ReactNode = <div data-testid="test-element" />;

const testConsumerContext = (
  children: (value: IConfirmedBookingsContext) => ReactNode
) => {
  return (
    <ConfirmedBookingsContext.Consumer>
      {children}
    </ConfirmedBookingsContext.Consumer>
  );
};

const setup = (children: ReactNode): JSX.Element => {
  return <TestComponent>{children}</TestComponent>;
};

describe('<ConfirmedBookingsProvider />', () => {
  test('should render the defined children element correctly', () => {
    const wrapper = render(setup(testElement));

    const children = wrapper.getByTestId('test-element');

    expect(children).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the specified bookings correctly after the click', () => {
    const context = testConsumerContext((value) => (
      <button
        onClick={() =>
          value.setConfirmedBookings((prevData) => [
            ...prevData,
            testBookingData
          ])
        }
        data-testid="booking-button"
      >
        {JSON.stringify(value.confirmedBookings)}
      </button>
    ));

    const wrapper = render(setup(context));

    const bookingButton = wrapper.getByTestId('booking-button');

    expect(bookingButton.textContent).toBe(`[]`);

    fireEvent.click(bookingButton);

    expect(bookingButton.textContent).toBe(
      `[${JSON.stringify(testBookingData)}]`
    );
    expect(wrapper).toMatchSnapshot();
  });
});
