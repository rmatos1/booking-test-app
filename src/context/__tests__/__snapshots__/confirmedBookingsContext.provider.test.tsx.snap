// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmedBookingsProvider /> should render the defined children element correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="test-element"
      />
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="test-element"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ConfirmedBookingsProvider /> should render the specified bookings correctly after the click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-testid="booking-button"
      >
        [{"id":"1","name":"Julia","email":"julia@teste.com","checkIn":"2023-12-19","checkOut":"2023-12-26","selectedBedroom":3,"qtyGuests":3,"totalPrice":450}]
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="booking-button"
    >
      [{"id":"1","name":"Julia","email":"julia@teste.com","checkIn":"2023-12-19","checkOut":"2023-12-26","selectedBedroom":3,"qtyGuests":3,"totalPrice":450}]
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
